### PR TITLE
support port-level traffic policy on inbound cluster

### LINKED ISSUE #22863: Support port-level traffic policy on inbound cluster
**Describe the feature request**

When a service contains multiple ports, pilot will build inbound clusters for them separately, and their traffic policy should be able to be configured independently.
In the current code, only the top-level Traffic Policy in the DestinationRule will be used to set the Circuit Breaker of the inbound clusters, and I think it would be better to prefer PortLevelSettings.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by kevin21th**
I think it would be better to prefer PortLevelSettings.
What I did is to prefer the Traffic Policy that matches the current inbound cluster port when applyConnectionPool. If there is no match, the top-level Traffic Policy would be used.

close #22863

### CODE REVIEW

**Comment by howardjohn** [comment]
/ok-to-test

This seems reasonable to me, but I'll let @rshriram sanity check this since I am not sure if there is any context for this not being implemented.

Thanks for the PR!

File: pilot/pkg/networking/core/v1alpha3/cluster_test.go [hunk 1]
```diff
@@ -25,6 +26,7 @@ import (
 	apiv2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
 	envoy_api_v2_auth "github.com/envoyproxy/go-control-plane/envoy/api/v2/auth"
 	apiv2_cluster "github.com/envoyproxy/go-control-plane/envoy/api/v2/cluster"
+	v2Cluster "github.com/envoyproxy/go-control-plane/envoy/api/v2/cluster"
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster_test.go:29]: duplicate

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 2]
```diff
@@ -489,10 +489,23 @@ func (configgen *ConfigGeneratorImpl) buildInboundClusterForPortOrUDS(pluginPara
 	cfg := pluginParams.Push.DestinationRule(pluginParams.Node, instance.Service)
 	if cfg != nil {
 		destinationRule := cfg.Spec.(*networking.DestinationRule)
+		var connectionPool *networking.ConnectionPoolSettings
 		if destinationRule.TrafficPolicy != nil {
+			for _, portTrafficPolicy := range destinationRule.TrafficPolicy.PortLevelSettings {
+				if portTrafficPolicy.GetPort().GetNumber() == uint32(instance.ServicePort.Port) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:495]: Try to reuse `SelectTrafficPolicyComponents`

File: pilot/pkg/networking/core/v1alpha3/cluster_test.go [hunk 3]
```diff
@@ -1885,6 +1887,187 @@ func TestBuildInboundClustersDefaultCircuitBreakerThresholds(t *testing.T) {
...
+
+	for _, cluster := range clusters {
+		g.Expect(cluster.CircuitBreakers).NotTo(BeNil())
+		if cluster.Name == "inbound|80||backend.default.svc.cluster.local" {
+			g.Expect(cluster.CircuitBreakers.Thresholds[0]).To(Equal(&exceptedCircuitBreakerThreshold))
+		}
+	}
+}
+
+func TestBuildInboundClustersCircuitBreakerThresholds(t *testing.T) {
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/cluster_test.go:1978]: nit: could you merge with the above test?

File: pilot/pkg/networking/core/v1alpha3/cluster_test.go (hunk 1 above)
Comment (kevin21th) [reply pilot/pkg/networking/core/v1alpha3/cluster_test.go]: fixed

File: pilot/pkg/networking/core/v1alpha3/cluster.go (hunk 2 above)
Comment (kevin21th) [reply pilot/pkg/networking/core/v1alpha3/cluster.go]: Thanks for the suggestion, indeed reusable

File: pilot/pkg/networking/core/v1alpha3/cluster_test.go (hunk 3 above)
Comment (kevin21th) [reply pilot/pkg/networking/core/v1alpha3/cluster_test.go]: merged

**Pull request merged**